fastparquet/writer.py

```diff
@@ -85,7 +86,7 @@ def find_type(data, convert=False, fixed_text=None):
     if dtype.name in typemap:
         type, converted_type, width = typemap[dtype.name]
         if type in revmap and convert:
-            out = data.values.astype(revmap[type])
+            out = data.values.astype(revmap[type], copy=False)
```

Comment:
Not necessarily, there are more pandas types than parquet ones. A copy should only be made if needed.